libs/api/domains/notifications/src/lib/notifications.model.ts

```diff
@@ -0,0 +1,143 @@
+import {
+  Field,
+  GraphQLISODateTime,
+  ID,
+  InputType,
+  ObjectType,
+  registerEnumType,
+  Int,
+} from '@nestjs/graphql'
+import { PageInfoDto } from '@island.is/nest/pagination'
+import { RenderedNotificationDtoStatusEnum } from '@island.is/clients/user-notification'
+
+registerEnumType(RenderedNotificationDtoStatusEnum, {
+  name: 'NotificationStatus',
+})
+
+@ObjectType('NotificationMetadata')
+export class NotificationMetadata {
+  @Field(() => GraphQLISODateTime)
+  sent!: Date
+
+  @Field(() => GraphQLISODateTime, { nullable: true })
+  updated?: Date
+
+  @Field(() => GraphQLISODateTime, { nullable: true })
+  created?: Date
+
+  @Field(() => GraphQLISODateTime, { nullable: true })
+  read?: Date
+
+  @Field(() => RenderedNotificationDtoStatusEnum)
+  status!: RenderedNotificationDtoStatusEnum
+}
+
+@ObjectType('NotificationSender')
+export class NotificationSender {
+  @Field()
+  name!: string
+
+  @Field({ nullable: true })
+  logo?: string
+}
+
+@ObjectType('NoticiationRecipient')
+export class NotificationRecipient {
+  @Field({ nullable: true })
+  nationalId?: string
+}
+
+@ObjectType('NotificationLink')
+export class NotificationLink {
+  @Field({ nullable: true })
+  uri?: string
+}
+
+@ObjectType('NotificationMessage')
+export class NotificationMessage {
+  @Field()
+  title!: string
+
+  @Field()
+  body!: string
+
+  @Field(() => NotificationLink)
+  link!: NotificationLink
+}
+
+@ObjectType('Notification')
+export class Notification {
+  @Field(() => Int)
+  id!: number
+
+  @Field(() => ID)
+  notificationId!: string
+
+  @Field(() => NotificationMetadata)
+  metadata!: NotificationMetadata
+
+  @Field(() => NotificationSender)
+  sender!: NotificationSender
+
+  @Field(() => NotificationRecipient)
+  recipient!: NotificationRecipient
+
+  @Field(() => NotificationMessage)
+  message!: NotificationMessage
+}
+
+@ObjectType('NotificationMessageCounts')
+export class NotificationMessageCounts {
+  @Field(() => Int, { nullable: true })
+  totalCount?: number
+
+  @Field(() => Int, { nullable: true })
+  unreadCount?: number
+}
+
+@InputType('NotificationsInput')
+export class NotificationsInput {
```

Comment:
There exists a helper function in @island.is/nest/graphql to add the pagination input properties. And the [pagination chapter](https://docs.devland.is/technical-overview/api-design-guide/pagination#pagination-query-parameters) in API Design Guides describes those fields to be named, `limit`, `before`, `after`.

```suggestion
export class NotificationsInput extends PaginationInput() {
```

Reply:
Yes! good call. Updated.